### ISSUE TITLE
sort() with {func} argument "F", sort on floats given as string 

### ISSUE THREAD

**Issue by Houl** [vim/vim#7857]
**Is your feature request about something that is currently impossible or hard to do? Please describe the problem.**
One can use `:sort n` to sort on numbers and `:sort f` to sort on floats in the buffer.
With given Lines (list of string), Lines can be sorted on numbers with `sort(Lines, 'N')`, but there is no way to do the same with contained floats.

**Describe the solution you'd like**
Please add `sort(Lines, 'F')` which should apply an implicit `str2float()` conversion on string items.
In other words: `sort(, 'F')` should be to `sort(, 'f')` what `sort(, 'N')` is to `sort(, 'n')`.

**Describe alternatives you've considered**
None, maybe check `:h sort() ` once more.
